packages/extension-ui/src/partials/NewAccountSelection.tsx

```diff
@@ -0,0 +1,281 @@
+// Copyright 2019-2023 @polkadot/extension-ui authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { ThemeProps } from '../types';
+
+import React, { useCallback, useContext, useEffect, useMemo, useState } from 'react';
+import styled from 'styled-components';
+
+import { AccountJson } from '@polkadot/extension-base/background/types';
+
+import plusIcon from '../assets/add.svg';
+import ribbon from '../assets/ribbon.svg';
+import { AccountContext, FaviconBox, Svg } from '../components';
+import Checkbox from '../components/Checkbox';
+import useTranslation from '../hooks/useTranslation';
+import Account from '../Popup/Accounts/Account';
+import AccountsTree from '../Popup/Accounts/AccountsTree';
+import { createGroupedAccountData } from '../util/createGroupedAccountData';
+
+interface Props extends ThemeProps {
+  className?: string;
+  url: string;
+  showHidden?: boolean;
+  newAccounts: Array<AccountJson> | [];
+}
+
+interface GroupedData {
+  [key: string]: AccountJson[];
+}
+
+const StyledCheckbox = styled(Checkbox)`
+  display: flex;
+  justify-content: flex-end;
+  margin-right: 8px;
+`;
+
+function NewAccountSelection({ className, newAccounts, showHidden = false, url }: Props): React.ReactElement<Props> {
+  const { t } = useTranslation();
+  const { accounts, hierarchy, selectedAccounts = [], setSelectedAccounts } = useContext(AccountContext);
+  const [isIndeterminate, setIsIndeterminate] = useState(false);
+
+  const { flattened, getParentName } = useMemo(() => createGroupedAccountData(hierarchy), [hierarchy]);
+  const allVisibleAccounts = useMemo(() => accounts.filter(({ isHidden }) => !isHidden), [accounts]);
+  const noAccountSelected = useMemo(() => selectedAccounts.length === 0, [selectedAccounts.length]);
```

Comment:
Does esLint requires usage of useMemo?
Overusing it is also bad for performance :) https://blog.logrocket.com/rethinking-hooks-memoization/
Especially in `noAccountSelected` or `areAllAccountsSelected` where the return is primitive it could be stripped down.
Plus if memoized values are not used directly in component but in another fn , the value of memoization is much lower.